src/ast/document-descriptor.ts

```diff
@@ -146,6 +148,9 @@ export class Document implements Feature {
   getByKind(kind: 'import'): Set<Import>;
   getByKind(kind: string): Set<Feature>;
   getByKind(kind: string): Set<Feature> {
+    if (this._featuresByKind) {
```

Comment:
Done!

Had to think about it a bit, and I documented the design constraints in the code with assertions (that should never trip in the current code, but which future mistaken modifications might)